accessories/base.js

```diff
@@ -139,6 +139,12 @@ module.exports = homebridge => {
           d.settings.hwinfo.hw_revision
         )
       }
+      
+      if (d.settings && d.settings.name) {
```

Comment:
Can you add `&& d.settings.name === this.name` here so that if a name has been specified in the config it will still override the device settings.